### ISSUE TITLE
`e2e_matmul_direct_*` tests not running in CI

### ISSUE THREAD

**Issue by GMNGeoffrey** [iree-org/iree#14133]
I can't figure out why, but the `e2e_matmul_direct_*` mostly aren't running in CI at all AFAICT. Looking at all the jobs from the most recent passing CI run (https://github.com/openxla/iree/actions/runs/5273187742), a few of them show up in the Bazel job, but that's it

- Bazel: https://github.com/openxla/iree/actions/runs/5273187742/jobs/9536311629
  - //tests/e2e/matmul:e2e_matmul_direct_f32_small_llvm-cpu_local-task
  - //tests/e2e/matmul:e2e_matmul_direct_f32_small_no_padding_llvm-cpu_local-task
  - //tests/e2e/matmul:e2e_matmul_direct_f32_small_ukernel_vmvx_local-task
  - //tests/e2e/matmul:e2e_matmul_direct_i8_small_llvm-cpu_local-task
  - //tests/e2e/matmul:e2e_matmul_direct_i8_small_ukernel_vmvx_local-task

I was slightly suspicious of my own https://github.com/openxla/iree/commit/2dddf02a12a2219fd80d457b5bf32dd155eb3b66 since it messed with exclusion tags, but looking at the job from before that, the same holds true: https://github.com/openxla/iree/actions/runs/5246578520

Another suspicious one was https://github.com/openxla/iree/commit/dc1684d9903fbe36d3f7884a02401c5d3f1815f2, but after that PR the tests actually *were* running: https://github.com/openxla/iree/actions/runs/4813700860/jobs/8570769114. So we've got a "regression" in that range. The investigation continues...

**Comment by GMNGeoffrey** [comment]
FYI @bjacob

**Comment by bjacob** [comment]
Good find! Are you continuing the bisection or need me to take this over?

**Comment by GMNGeoffrey** [comment]
Bisection revealed #13915 was the culprit. Hooray for `git bisect`, which made this easier even though I was still manually clicking around in logs to evaluate the commits

**Comment by bjacob** [comment]
Ha, ok, thanks very much for the bisection. I'll work on the fix now. self-assigning.

**Comment by GMNGeoffrey** [comment]
This sort of thing has happened a couple times. I'm trying to think of a way we could catch it. Something like parsing the CI logs and looking for changes in the names of tests run, but I think that would be super noisy

**Comment by GMNGeoffrey** [comment]
FYI, it looks like these tests actually just aren't configured at all:

```
gcmn@ct 2023-06-15 11:51 [iree] [hi] ~/src/iree$ ctest --test-dir ~/build/iree/default -N -R e2e_matmul
Internal ctest changing into directory: /usr/local/google/home/gcmn/build/iree/default
Test project /usr/local/google/home/gcmn/build/iree/default
  Test   #41: iree/tests/e2e/matmul/e2e_matmul_mmt4d_i8_small_ukernel_llvm-cpu_local-task_avx2_fma
  Test   #42: iree/tests/e2e/matmul/e2e_matmul_mmt4d_i8_small_ukernel_llvm-cpu_local-task_avx512_base
  Test   #43: iree/tests/e2e/matmul/e2e_matmul_mmt4d_i8_small_ukernel_llvm-cpu_local-task_avx512_vnni
  Test   #44: iree/tests/e2e/matmul/e2e_matmul_mmt4d_i8_large_ukernel_llvm-cpu_local-task_avx2_fma
  Test   #45: iree/tests/e2e/matmul/e2e_matmul_mmt4d_i8_large_ukernel_llvm-cpu_local-task_avx512_base
  Test   #46: iree/tests/e2e/matmul/e2e_matmul_mmt4d_i8_large_ukernel_llvm-cpu_local-task_avx512_vnni
  Test   #47: iree/tests/e2e/matmul/e2e_matmul_mmt4d_f32_small_ukernel_llvm-cpu_local-task_avx2_fma
  Test   #48: iree/tests/e2e/matmul/e2e_matmul_mmt4d_f32_small_ukernel_llvm-cpu_local-task_avx512_base
  Test   #49: iree/tests/e2e/matmul/e2e_matmul_mmt4d_f32_large_ukernel_llvm-cpu_local-task_avx2_fma
  Test   #50: iree/tests/e2e/matmul/e2e_matmul_mmt4d_f32_large_ukernel_llvm-cpu_local-task_avx512_base
  Test #1311: iree/builtins/ukernel/tools/e2e_matmul_benchmark_test

Total Tests: 11

gcmn@ct 2023-06-15 11:51 [iree] [hi] ~/src/iree$ ctest --test-dir ~/build/iree/default -N -R e2e_matmul_direct
Internal ctest changing into directory: /usr/local/google/home/gcmn/build/iree/default
Test project /usr/local/google/home/gcmn/build/iree/default

Total Tests: 0
```

**Comment by bjacob** [comment]
Fix: #14137

**Comment by bjacob** [comment]
Since this was a cmake-time bug, this could be caught by cmake logic, say we could require certain targets to exist at the end of cmake.
